### PR TITLE
Clean out `rules` "slots"

### LINKED ISSUE #105: [Rules] Flesh out `pop_slot_item_with_rules`
I noted that in this PR there's an area with a commented out function, `pop_slot_item_with_rules`

As we hustle to get our MVP ready for Cosmoverse, I totally get it! I thought I'd make a ticket to ensure we don't lose track of finishing that function.

See this comment, which is around the area of the function:
https://github.com/CronCats/cw-croncat/pull/104/files#r975953271

### PR DESCRIPTION
**Pull request by Buckram123**
Instead of "slots" of `time` to `[hash]` or `block` to `[hash]`: 
```rust 
Map<'a, u64, Vec<Vec<u8>>>
``` 
for rules we instead have a strict map of hash to time or block:
```rust
    pub time_map_rules: Map<'a, &'a [u8], u64>,
    pub block_map_rules: Map<'a, &'a [u8], u64>,
```
At the high level we have 3 operations for existing `task_hash`
- [x] Check if `hash` ready
- [x] Update time/block for this `hash`
- [x] Remove `hash`

Closes #105

### CODE REVIEW

**Comment by Buckram123** [comment]
Noticed we don't have update `time`/`block` for `task_hash` for rescheduling, fix and test in progress!

File: contracts/cw-croncat/src/tasks.rs [hunk 1]
```diff
@@ -518,6 +503,29 @@ impl<'a> CwCroncat<'a> {
         }
     }
 
+    fn pop_task_with_rule(
```
Comment (deveusss) [inline contracts/cw-croncat/src/tasks.rs:506]: great!)

**Review by deveusss — APPROVED** [review]

**Comment by mikedotexe** [comment]
Hmm, is this the DAO DAO version issue we recently saw?

Screen Shot 2022-10-21 at 11 16 07 AM

**Comment by Buckram123** [comment]
@mikedotexe just `Cargo.lock`, automerge fixed it

**Pull request merged**